index.js

```diff
@@ -6,6 +6,60 @@ const KEYCODE = {
 }
 
 const state = new Map()
+// when container or children get focus
+const onFocusin = e => {
+  const {currentTarget: rover} = e
+  if (state.get('last_rover') == rover) return
+  if (state.has(rover)) {
+    activate(rover, state.get(rover).active)
+    state.set('last_rover', rover)
+  }
+}
+const onKeydown = e => {
+  const {currentTarget: rover} = e
+  switch (e.keyCode) {
+    case KEYCODE[isRtl ? 'LEFT' : 'RIGHT']:
+    case KEYCODE.DOWN:
+      e.preventDefault()
+      focusNextItem(rover)
+      break
+    case KEYCODE[isRtl ? 'RIGHT' : 'LEFT']:
+    case KEYCODE.UP:
+      e.preventDefault()
+      focusPreviousItem(rover)
+      break
+  }
+}
+const mo = new MutationObserver((mutationList, observer) =>{
+  const stateElementsSet = new Set();
+  state.forEach((v,k) =>{
+    stateElementsSet.add(k.classList[0])
+  } )
+  mutationList
+    .filter(x => x.removedNodes.length > 0)
+    .forEach(mutation => {
+      mutation.removedNodes.forEach(removedEl => {
+      if (removedEl.nodeType !== 1) return //only elements
```

Comment:
turn this into a filter that runs before the forEach?

Reply:
Well removedNodes is a Nodelist not an Array so it doesn't have a filter method. I'd have to create an Array from it.
Do you have a preferred way to do this. My first instinct would be to use the spread operator. I've tried avoiding getting to modern with the EcmaScript since I didn't know how low of a version this is suppose to support

Reply:
And the feedback is very much appreciated.

Reply:
i spread or use `Array.from()`. but yeah, you're right that it's a bit much to cast it to an array type just to filter out 1 element. maybe let's stick with what you got here!?